File: c#/7_kyu/Factorial.cs
using System;

public static class Kata
{
    public static int Factorial(int n)
    {
        if (n < 0 || n > 12)
            throw new ArgumentOutOfRangeException();

        var output = 1;
        var multiplier = n;
        while (multiplier > 1)
        {
            output *= multiplier;
            multiplier--;
        }
        return output;
    }
}

// https://www.codewars.com/kata/54ff0d1f355cfd20e60001fc/train/csharp

File: CodeWars/java/7_kyu/Factorial/Factorial.java
// https://www.codewars.com/kata/57a049e253ba33ac5e000212/train/java

public class Factorial {
    public static long factorial(int n) {
        if (n == 0)
            return 1;
        long result = n;
        for (int i = n - 1; i > 1; i--) {
            result *= i;
        }
        return result;
    }
}
